### PR TITLE
shops go brrrrrrrrr

### PR DESCRIPTION
**Pull request by Grumblin**
created 'specialty shop' for teleportation crystal
updated the survivalist shop
added the item and handler for the teleportation crystal recipe item

### CODE REVIEW

File: denizen_scripts/survival/repo-link/dragon/teleportation_crystal_recipe_item.dsc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+teleportation_crystal_recipe_handler:
+    type: world
+    events:
+        on player right clicks with:recipe_item_name:
+        - if !<player.has_flag[teleportation_recipe]>
+            - flag teleportation_recipe
+            - take scriptname:teleportation_crystal_recipe_item
+        - else
+            - determine cancelled
+            - narrate "You can't use this item again"
```
Comment (BehrRiley) [inline denizen_scripts/survival/repo-link/dragon/teleportation_crystal_recipe_item.dsc]: `determine cancelled` will end the queue; you should ideally swap this line and the last line, or specify the `passively` argument.
A new-line at the end of the file is ideal here as well.
```suggestion
            - narrate "You can't use this item again"
            - determine cancelled

```

File: denizen_scripts/survival/repo-link/dragon/teleportation_crystal_recipe_item.dsc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    material: paper
+    display name: <&a>Recipe: Teleportation Crystal
+    mechanisms:
+        nbt: teleportgoesweeeee
+
+teleportation_crystal_recipe_handler:
+    type: world
+    events:
+        on player right clicks with:recipe_item_name:
+        - if !<player.has_flag[teleportation_recipe]>
```
Comment (ChrispyMC) [inline denizen_scripts/survival/repo-link/dragon/teleportation_crystal_recipe_item.dsc:12]:
```suggestion
        - if !<player.has_flag[teleportation_recipe]>:
```

File: denizen_scripts/survival/repo-link/shops/specialty_shop.dsc [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        - trigger name:click state:true
+        on click:
+        - inject specialtyInventoryInject
+
+specialtyInventoryInject:
+    type: task
+    script:
+    - define inventory <inventory[specialtyInventory]>
+    - give "<item[player_head].with[skull_skin=<player.skull_skin>;display_name=<&e><&o><player.display_name>;lore=<&a>Money: <&e><player.money>]>" to:<[inventory]> slot:5
+    - inventory open d:<[inventory]>
```
Comment (BehrRiley) [inline denizen_scripts/survival/repo-link/shops/specialty_shop.dsc:14]: If this is the only time this script is used, it's a better idea to just consolidate the script within the single script container.
```suggestion
        - define inventory <inventory[specialtyInventory]>
        - give "<item[player_head].with[skull_skin=<player.skull_skin>;display_name=<&e><&o><player.display_name>;lore=<&a>Money: <&e><player.money>]>" to:<[inventory]> slot:5
        - inventory open d:<[inventory]>
```

File: denizen_scripts/survival/repo-link/shops/specialty_shop.dsc [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                    - else:
+                        - narrate "<&c>You don't have the specified item!"
+            - stop
+
+specialtyData:
+    type: data
+    Items:
+        Teleportation_Shard:
+            Item: <item[teleportation_shard]>
+            Prices: 500/250
```
Comment (BehrRiley) [inline denizen_scripts/survival/repo-link/shops/specialty_shop.dsc:68]: A new-line at the end of the file is ideal here.
```suggestion
            Prices: 500/250

```

File: denizen_scripts/survival/repo-link/dragon/teleportation_crystal_recipe_item.dsc (hunk 1 above)
Comment (ChrispyMC) [inline denizen_scripts/survival/repo-link/dragon/teleportation_crystal_recipe_item.dsc]:
```suggestion
            - narrate "You can't use this item again"

```

Comment (ChrispyMC) [reply denizen_scripts/survival/repo-link/dragon/teleportation_crystal_recipe_item.dsc]: There should be a new line at the end of the file.

File: denizen_scripts/survival/repo-link/shops/specialty_shop.dsc [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                        - playsound <player> sound:ENTITY_VILLAGER_NO volume:0.6 pitch:1.4
+                - case sell:
+                    - if <player.inventory.contains.scriptname[<context.item.nbt[key]>]>:
+                        - playsound <player> sound:UI_BUTTON_CLICK volume:0.6 pitch:1.4
+                        - take slot:<player.inventory.find.scriptname[<context.item.nbt[key]>]>
+                        - money give quantity:<script[specialtyData].data_key[Items.<context.item.nbt[key]>.Prices].after[/]>
+                        - inventory set "o:<item[player_head].with[skull_skin=<player.skull_skin>;display_name=<&e><&o><player.display_name>;lore=<&a>Money: <&e><player.money>]>" to:<player.open_inventory> slot:5
+                    - else:
+                        - narrate "<&c>You don't have the specified item!"
+            - stop
```
Comment (BehrRiley) [inline denizen_scripts/survival/repo-link/shops/specialty_shop.dsc:61]: `stop` is not needed when it's the end of the script.

File: denizen_scripts/survival/repo-link/shops/specialty_shop.dsc (hunk 4 above)
Comment (ChrispyMC) [inline denizen_scripts/survival/repo-link/shops/specialty_shop.dsc:68]:
```suggestion
            Prices: 500/250

```
There should be a new line at the end of the file.

File: denizen_scripts/survival/repo-link/shops/specialty_shop.dsc [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+                        - inventory set "o:<item[player_head].with[skull_skin=<player.skull_skin>;display_name=<&e><&o><player.name>;lore=<&a>Money: <&e><player.money>]>" to:<player.open_inventory> slot:5
+                    - else:
+                        - narrate "<&c>You don't have enough money"
+                        - playsound <player> sound:ENTITY_VILLAGER_NO volume:0.6 pitch:1.4
+                - case sell:
+                    - if <player.inventory.contains.scriptname[<context.item.nbt[key]>]>:
+                        - playsound <player> sound:UI_BUTTON_CLICK volume:0.6 pitch:1.4
+                        - take slot:<player.inventory.find.scriptname[<context.item.nbt[key]>]>
+                        - money give quantity:<script[specialtyData].data_key[Items.<context.item.nbt[key]>.Prices].after[/]>
+                        - inventory set "o:<item[player_head].with[skull_skin=<player.skull_skin>;display_name=<&e><&o><player.display_name>;lore=<&a>Money: <&e><player.money>]>" to:<player.open_inventory> slot:5
```
Comment (BehrRiley) [inline denizen_scripts/survival/repo-link/shops/specialty_shop.dsc:58]: `to:` is an invalid argument for the `inventory` command; consider either `give` or `inventory set destination:`; `d:` for short
```suggestion
                        - inventory set "o:<item[player_head].with[skull_skin=<player.skull_skin>;display_name=<&e><&o><player.display_name>;lore=<&a>Money: <&e><player.money>]>" d:<player.open_inventory> slot:5
```
```suggestion
                        - give "<item[player_head].with[skull_skin=<player.skull_skin>;display_name=<&e><&o><player.display_name>;lore=<&a>Money: <&e><player.money>]>" to:<player.open_inventory> slot:5
```

File: denizen_scripts/survival/repo-link/shops/specialty_shop.dsc [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            - inventory set d:<player.open_inventory> o:<item[red_wool].with[nbt=action/sell|key/<context.item.nbt[item]>;display_name=<&c>Sell<&sp>1;lore=<&a>Price:<&sp><script[specialtyData].data_key[Items.<context.item.nbt[item]>.Prices].after[/]>]> slot:49
+            - stop
+        - if <context.item.has_nbt[action]>:
+            - choose <context.item.nbt[action]>:
+                - case buy:
+                    - if <player.money> >= <script[specialtyData].data_key[Items.<context.item.nbt[key]>.Prices].before[/]>:
+                        - playsound <player> sound:UI_BUTTON_CLICK volume:0.6 pitch:1.4
+                        - give <script[specialtyData].parsed_key[Items.<context.item.nbt[key]>.Item]>
+                        - money take quantity:<script[specialtyData].data_key[Items.<context.item.nbt[key]>.Prices].before[/]>
+                        - inventory set "o:<item[player_head].with[skull_skin=<player.skull_skin>;display_name=<&e><&o><player.name>;lore=<&a>Money: <&e><player.money>]>" to:<player.open_inventory> slot:5
```
Comment (BehrRiley) [inline denizen_scripts/survival/repo-link/shops/specialty_shop.dsc:49]: `to:` is an invalid argument for the `inventory` command; consider either `give` or `inventory set destination:`; `d:` for short
```suggestion
                        - inventory set "o:<item[player_head].with[skull_skin=<player.skull_skin>;display_name=<&e><&o><player.name>;lore=<&a>Money: <&e><player.money>]>" d:<player.open_inventory> slot:5
```
```suggestion
                        - give "<item[player_head].with[skull_skin=<player.skull_skin>;display_name=<&e><&o><player.name>;lore=<&a>Money: <&e><player.money>]>" to:<player.open_inventory> slot:5
```

**Review by BehrRiley — CHANGES_REQUESTED** [review]

File: denizen_scripts/survival/repo-link/dragon/teleportation_crystal_recipe_item.dsc [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+teleportation_crystal_recipe_handler:
+    type: world
+    events:
+        on player right clicks with:recipe_item_name:
+        - if !<player.has_flag[teleportation_recipe]>:
+            - flag teleportation_recipe
+            - take scriptname:teleportation_crystal_recipe_item
+        - else
+            - determine cancelled
+            - narrate "You can't use this item again"
```
Comment (BehrRiley) [inline denizen_scripts/survival/repo-link/dragon/teleportation_crystal_recipe_item.dsc:17]: `determine cancelled` will end the queue; you should ideally swap this line and the last line, or specify the `passively` argument.
```suggestion
            - narrate "You can't use this item again"
            - determine cancelled
```

**Review by BehrRiley — APPROVED** [review]

**Pull request merged**